metacat-main/src/main/java/com/netflix/metacat/main/manager/CatalogManager.java

```diff
@@ -114,16 +124,22 @@ private void loadCatalog(final File file) throws Exception {
         log.info("-- Loading catalog {} --", file);
         final Map<String, String> properties = new HashMap<>(this.loadProperties(file));
 
-        final String connectorType = properties.remove("connector.name");
+        final String connectorType = properties.remove(MetacatCatalogConfig.Keys.CONNECTOR_NAME);
         Preconditions.checkState(
             connectorType != null,
             "Catalog configuration %s does not contain connector.name",
             file.getAbsoluteFile()
         );
 
-        final String catalogName = Files.getNameWithoutExtension(file.getName());
-        final ConnectorContext connectorContext = new ConnectorContext(catalogName, config, registry, properties);
-        this.connectorManager.createConnection(catalogName, connectorType, connectorContext);
+        // Catalog shard name should be unique. Usually the catalog name is same as the catalog shard name.
+        // If multiple catalog property files refer the same catalog name, then there will be multiple shard names
+        // with the same catalog name.
+        final String catalogShardName = Files.getNameWithoutExtension(file.getName());
+        // If catalog name is not specified, then use the catalog shard name.
+        final String catalogName = properties.getOrDefault(MetacatCatalogConfig.Keys.CATALOG_NAME, catalogShardName);
+        final ConnectorContext connectorContext =
+            new ConnectorContext(catalogName, catalogShardName, connectorType, config, registry, properties);
+        this.connectorManager.createConnection(connectorContext);
         log.info("-- Added catalog {} using connector {} --", catalogName, connectorType);
     }
```

Comment:
are we allowing one file having multiple shards configs? Also, can we log the shard explicitly? 
`log.info("-- Added catalog {} using connector {} --", catalogName, connectorType);`